src/main/java/com/epam/university/java/core/task028/Task028Impl.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.university.java.core.task028;
+
+import java.util.stream.Stream;
+
+/**
+ * Created by ilya on 03.10.17.
+ */
+public class Task028Impl implements Task028 {
+
+    private int varCounter;
+    private int[] powers;
```

Comment:
May be it's better to use collections?)

Reply:
Yes, and you know that I can, just here I had been doing some optimization last time, and think that this is faster way, but I don't sure very well.